src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -809,9 +806,9 @@ internal sealed class AwaitableSocketAsyncEventArgs : SocketAsyncEventArgs, IVal
         {
             internal static readonly AwaitableSocketAsyncEventArgs Reserved = new AwaitableSocketAsyncEventArgs() { _continuation = null };
             /// <summary>Sentinel object used to indicate that the operation has completed prior to OnCompleted being called.</summary>
-            private static readonly Action<object> s_completedSentinel = state => throw new Exception(nameof(s_completedSentinel));
+            private static readonly Action<object> s_completedSentinel = new Action<object>(state => throw new Exception(nameof(s_completedSentinel)));
             /// <summary>Sentinel object used to indicate that the instance is available for use.</summary>
-            private static readonly Action<object> s_availableSentinel = state => throw new Exception(nameof(s_availableSentinel));
+            private static readonly Action<object> s_availableSentinel = new Action<object>(state => throw new Exception(nameof(s_availableSentinel)));
```

Comment:
These shouldn't make any difference, but I'm being explicit to ensure the compiler isn't playing any tricks around folding multiple identical delegates together and defeating object identity checks.